base/float.jl

```diff
@@ -501,15 +501,41 @@ for Ti in (Int64,UInt64,Int128,UInt128)
         end
     end
 end
+==(x::Float16, y::Union{Int128,UInt128}) = Float64(x)==Float64(y)
```

Comment:
This seems problematic since not every `y` can be represented as a `Float64`.

Reply:
Why? That was addressed in the previous PR discussion.